src/modules/launcher/Plugins/Microsoft.Plugin.Program/Programs/Win32Program.cs

```diff
@@ -885,41 +802,36 @@ private static string GetProgramPathFromRegistrySubKeys(RegistryKey root, string
             }
             catch (Exception e) when (e is SecurityException || e is UnauthorizedAccessException)
             {
-                ProgramLogger.Exception($"|Permission denied when trying to load the program from {path}", e, MethodBase.GetCurrentMethod().DeclaringType, path);
+                ProgramLogger.Warn($"|Permission denied when trying to load the program from {path}", e, MethodBase.GetCurrentMethod().DeclaringType, path);
 
                 return string.Empty;
             }
         }
 
-        private static Win32Program GetProgramFromPath(string path)
-        {
-            if (string.IsNullOrEmpty(path))
-            {
-                return new Win32Program();
-            }
-
-            path = Environment.ExpandEnvironmentVariables(path);
-
-            if (!File.Exists(path))
-            {
-                return new Win32Program();
-            }
-
-            var entry = CreateWin32Program(path);
-            entry.ExecutableName = Path.GetFileName(path);
-
-            return entry;
-        }
+        private static string ExpandEnvironmentVariables(string path) =>
+            path != null
+                ? Environment.ExpandEnvironmentVariables(path)
+                : null;
 
         // Overriding the object.GetHashCode() function to aid in removing duplicates while adding and removing apps from the concurrent dictionary storage
         public override int GetHashCode()
-        {
-            return RemoveDuplicatesComparer.Default.GetHashCode(this);
-        }
+            => Win32ProgramEqualityComparer.Default.GetHashCode(this);
 
         public override bool Equals(object obj)
+            => obj is Win32Program win && Win32ProgramEqualityComparer.Default.Equals(this, win);
+
+        private class Win32ProgramEqualityComparer : IEqualityComparer<Win32Program>
         {
-            return obj is Win32Program win && RemoveDuplicatesComparer.Default.Equals(this, win);
+            public static readonly Win32ProgramEqualityComparer Default = new Win32ProgramEqualityComparer();
+
+            public bool Equals(Win32Program app1, Win32Program app2)
+                => app1 != null
+                   && app2 != null
+                   && (app1.Name?.ToUpperInvariant(), app1.ExecutableName?.ToUpperInvariant(), app1.FullPath?.ToUpperInvariant())
+                    .Equals((app2.Name?.ToUpperInvariant(), app2.ExecutableName?.ToUpperInvariant(), app2.FullPath?.ToUpperInvariant()));
```

Comment:
if `app1 == null` and `app2 == null` then the function returns false. Shouldn't it return true?